packages/postcss-merge-longhand/src/lib/decl/borders.js

```diff
@@ -20,6 +20,7 @@ const { isValidWsc } = require('../validateWsc.js');
 
 const wsc = ['width', 'style', 'color'];
 const defaults = ['medium', 'none', 'currentcolor'];
+const colorMightRequireFallback = /hsla\(|rgba\(|color\(/i;
```

Comment:
Looks good, but there are more color functions https://www.w3.org/TR/css-color-4/#color-syntax

Reply:
Whoops, you're right. There's also an `isColor` function somewhere in the `postcss-merge-longhand` plugin, but updating that one has no effect on this bug, so I've left it alone as I don't understand what it does. I'll add the color functions to the regex.

Reply:
I've added more color functions. To be consistent with how other features are handled, we could check if the browsers in browserslist support a color before merging it, on the other hand if the user has added a fallback, we could assume they know the color does not work on all target browsers. To do the merging properly we would need more precise parsing of the color values. For example, according to CanIUse, even IE6 supports `rgb()` with integer values, but no IE version supports `rgb` with float values.

Reply:
Yes, you are right, we can fix it in the next PR